packages/grafana-data/src/transformations/transformers/convertFieldType.ts

```diff
@@ -150,6 +150,10 @@ function fieldToNumberField(field: Field): Field {
     if (valuesAsStrings && toBeConverted != null) {
       // some numbers returned from datasources have commas
       // strip the commas, coerce the string to a number
+      if (typeof toBeConverted !== 'string') {
+        toBeConverted = String(toBeConverted);
+      }
+
```

Comment:
There are situations where values are not string in a string field type. e.g: when drag/dropping and csv the dataframe string fields can have both numeric and strings, thus the need for a conversion here. Not sure if fixing it in the convert transformer is the way to go, I think a better approach is to try to convert values when importing, maybe.

Reply:
refactored this a bit

Reply:
I'm okay with this. Would be nice to have a test for your use case as well.